### PR TITLE
GE Signa Parameters Fix #583

### PR DESCRIPTION
**Pull request by pwadhwa351**
GE Signa Scanner Parameters corrected in GEHDF5Wrapper.cxx and Scanner.cxx

### CODE REVIEW

**Comment by pw351** [comment]
I am calculating DOI using inner radius (or physical radius of the scanner). The value is not present in header, what would you suggest?

Also  max_num_of_non_arc_corrected_bins is not really present in listmode file header (value is 0) but is present in the rdf file, I guess!

**Comment by pwadhwa351** [comment]
I am calculating DOI using inner radius (or physical radius of the scanner). The value is not present in header, what would you suggest?

Also max_num_of_non_arc_corrected_bins is not really present in listmode file header (value is 0) but is present in the rdf file, I guess!

**Comment by pwadhwa351** [comment]
@KrisThielemans I think this is done!

File: src/IO/GEHDF5Wrapper.cxx [hunk 1]
```diff
@@ -304,51 +302,41 @@ Succeeded GEHDF5Wrapper::initialise_scanner_from_HDF5()
...
-                                         num_transaxial_blocks_per_bucket,
-                                         num_axial_crystals_per_block,
-                                         num_transaxial_crystals_per_block,
-                                         num_axial_crystals_per_singles_unit,
-                                         num_transaxial_crystals_per_singles_unit,
-                                         num_detector_layers,
-                                         energy_resolution,
-                                         reference_energy));
+    shared_ptr<Scanner> scanner_sptr(Scanner::get_scanner_from_name(read_str_scanner));
+    scanner_sptr->list_names(read_str_scanner);
```
Comment (KrisThielemans) [inline src/IO/GEHDF5Wrapper.cxx]:
```suggestion
```
`list_names()` returns the names. Here you have set it above.

File: src/IO/GEHDF5Wrapper.cxx [hunk 2]
```diff
@@ -304,51 +302,41 @@ Succeeded GEHDF5Wrapper::initialise_scanner_from_HDF5()
...
-                                         num_axial_blocks_per_bucket,
-                                         num_transaxial_blocks_per_bucket,
-                                         num_axial_crystals_per_block,
-                                         num_transaxial_crystals_per_block,
-                                         num_axial_crystals_per_singles_unit,
-                                         num_transaxial_crystals_per_singles_unit,
-                                         num_detector_layers,
-                                         energy_resolution,
-                                         reference_energy));
+    shared_ptr<Scanner> scanner_sptr(Scanner::get_scanner_from_name(read_str_scanner));
```
Comment (KrisThielemans) [inline src/IO/GEHDF5Wrapper.cxx:316]:
```suggestion
    shared_ptr<Scanner> scanner_sptr(Scanner::get_scanner_from_name(read_str_scanner));
    if (is_null_ptr(scanner_sptr))
       error("Scanner read from RDF file is " + read_str_scanner + ", but this is not supported yet");
    if (scanner_sptr->get_type() != Scanner::PETMR_Signa)
      warning("Scanner read from RDF file is " + read_str_scanner + ", but this code is only tested for the Signa PET/MR");
       
```

File: src/IO/GEHDF5Wrapper.cxx [hunk 3]
```diff
@@ -259,6 +259,8 @@ Succeeded GEHDF5Wrapper::initialise_scanner_from_HDF5()
     H5::DataSet dataset = file.openDataSet("/HeaderData/ExamData/scannerDesc");
     dataset.read(read_str_scanner,vlst);
 
+    if(read_str_scanner == "SIGNA PET/MR")
+    {
```
Comment (KrisThielemans) [inline src/IO/GEHDF5Wrapper.cxx]:
```suggestion
```
let's always read it. checks are below

File: src/IO/GEHDF5Wrapper.cxx [hunk 4]
```diff
@@ -304,51 +302,41 @@ Succeeded GEHDF5Wrapper::initialise_scanner_from_HDF5()
...
+    scanner_sptr->set_num_detector_layers(num_detector_layers);
+    scanner_sptr->set_energy_resolution(energy_resolution);
+    scanner_sptr->set_reference_energy(reference_energy);
 
     return Succeeded::yes;
+    }
+    else
+    {
+        return Succeeded::no;
+    }
```
Comment (KrisThielemans) [inline src/IO/GEHDF5Wrapper.cxx]:
```suggestion
```
removed, as `if` above removed

**Review by KrisThielemans — CHANGES_REQUESTED** [review]

**Review by KrisThielemans — APPROVED** [review]

**Comment by KrisThielemans** [comment]
Thanks @pwadhwa351 !

@AnderBiguri could you merge this? I cannot.

**Comment by pwadhwa351** [comment]
Ah! Thanks for the merge, one last thing was left! Basically we need to add #include "stir/is_null_ptr.h" in GEHDF5Wrapper.cxx!

**Comment by KrisThielemans** [comment]
done!

Could you share the uncompressed sino with me as well? I don't think I have it.

**Pull request merged**